semgrep-core/src/engine/Specialize_formula.ml

```diff
@@ -31,11 +40,22 @@ let selector_equal s1 s2 = s1.mvar = s2.mvar
 (* Converter *)
 (*****************************************************************************)
 
+(* return list of "positive" x list of Not x list of Conds *)
+let split_and :
+    R.formula list -> R.formula list * R.formula list * R.metavar_cond list =
+ fun xs ->
+  xs
+  |> Common.partition_either3 (fun e ->
+         match e with
+         | R.Not (_, f) -> Middle3 f
+         | R.Leaf (R.MetavarCond (_, c)) -> Right3 c
+         | _ -> Left3 e)
```

Comment:
imo this is worth enumerating, so that a reader can see what is considered positive. Also, to guard against the off chance we add another negative term.

Reply:
I agree!